joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstCreatorHelper.scala

```diff
@@ -91,16 +85,97 @@ trait AstCreatorHelper { this: AstCreator =>
-      case CompositeLit => s"${jsonNode.obj(ParserKeys.Type)(ParserKeys.Elt)(ParserKeys.Name).str}[]"
-      case Ellipsis     => "..." + jsonNode.obj(ParserKeys.Elt)(ParserKeys.Name).str
+      case Ident =>
+        val typeNameForcode = nodeInfo.json(ParserKeys.Name).str
+        val fullName        = generateTypeFullName(typeNameForcode)
+        (fullName, typeNameForcode)
       case SelectorExpr =>
-        jsonNode.obj(ParserKeys.X)(ParserKeys.Name).str + "." + jsonNode.obj(ParserKeys.Sel)(ParserKeys.Name).str
-      case _ => Defines.anyTypeName
+        val typeNameForcode =
+          s"${nodeInfo.json(ParserKeys.X)(ParserKeys.Name).str}.${nodeInfo.json(ParserKeys.Sel)(ParserKeys.Name).str}"
+        val fullName = generateTypeFullName(
+          typeName = nodeInfo.json(ParserKeys.Sel)(ParserKeys.Name).str,
+          aliasName = Some(nodeInfo.json(ParserKeys.X)(ParserKeys.Name).str)
+        )
+        (fullName, typeNameForcode)
+      case InterfaceType =>
+        val typeNameForcode = "interface{}"
+        val fullName        = generateTypeFullName(typeNameForcode)
+        (fullName, typeNameForcode)
+      case _ =>
+        val fullName = generateTypeFullName()
+        (fullName, fullName)
+    }
+  }
+
+  private def internalArrayTypeHandler(nodeInfo: ParserNodeInfo): (String, String) = {
+    nodeInfo.node match {
+      case ArrayType =>
+        val (fullName, typeNameForcode) = internalTypeFullName(createParserNodeInfo(nodeInfo.json(ParserKeys.Elt)))
+        (s"[]${fullName}", s"[]${typeNameForcode}")
+      case CompositeLit =>
+        val (fullName, typeNameForcode) = internalTypeFullName(
+          createParserNodeInfo(nodeInfo.json(ParserKeys.Type)(ParserKeys.Elt))
+        )
+        (s"[]${fullName}", s"[]${typeNameForcode}")
+      case _ =>
+        val (fullName, typeNameForcode) = internalTypeFullName(nodeInfo)
+        (fullName, typeNameForcode)
+    }
+  }
+
+  private def internalStarExpHandler(nodeInfo: ParserNodeInfo): (String, String, String) = {
+    nodeInfo.node match {
+      case StarExpr =>
+        // TODO: Need to handle pointer to pointer use case.
+        val (fullName, typeNameForcode) = internalArrayTypeHandler(createParserNodeInfo(nodeInfo.json(ParserKeys.X)))
+        (s"*${fullName}", s"*${typeNameForcode}", EvaluationStrategies.BY_SHARING)
+      case _ =>
+        val (fullName, typeNameForcode) = internalArrayTypeHandler(nodeInfo)
+        (fullName, typeNameForcode, EvaluationStrategies.BY_VALUE)
+    }
+  }
+
+  protected def processTypeInfo(jsonNode: Value): (String, String, Boolean, String) = {
+    val nodeInfo = createParserNodeInfo(jsonNode)
+    nodeInfo.node match {
+      case ArrayType =>
+        val (fullName, typeNameForcode, evaluationStrategy) = internalStarExpHandler(
+          createParserNodeInfo(jsonNode.obj(ParserKeys.Elt))
+        )
+        (s"[]${fullName}", s"[]${typeNameForcode}", false, evaluationStrategy)
```

Comment:
You don't need the braces here
```suggestion
        (s"[]$fullName", s"[]$typeNameForcode", false, evaluationStrategy)
```

Reply:
done